src/components/pie-chart/pie-chart.js

```diff
@@ -1,32 +1,71 @@
 import template from "./pie-chart.html";
 import Chart from "chart.js";
 
+import { CurrencyConverterUAHService } from "../../services/currency-converter-uah";
+import { ColorGeneratorService } from "../../services/color-generator";
+
 export class PieChartComponent {
   constructor(mountPoint) {
     this.mountPoint = mountPoint;
+    this.dataset = { tags: [], amounts: [], colors: [] };
   }
 
   querySelectors() {
-    this.pieChartCtx = this.mountPoint.querySelector(".chart__visual");
+    this.chartCtx = this.mountPoint.querySelector(".chart__visual");
   }
 
-  mount() {
-    this.mountPoint.innerHTML = template();
-    this.querySelectors();
-    this.createPieChart();
+  update(data) {
+    if (data.type === "+") {
+      return;
+    }
+    this.dataset = this.addCurrData(this.dataset, data);
+    this.drawChanged();
+  }
+
+  createFromList(list) {
+    let expenceList = list.filter(item => item.type === "-").reverse();
+    this.dataset = expenceList.reduce(this.addCurrData, this.dataset);
+    this.drawChanged();
+  }
+
+  addCurrData(accum, item) {
+    const data = Object.assign({}, accum);
```

Comment:
this does not help here  because `Object.assign` makes shallow copy, not deep -- possible solution is to use `_.cloneDeep` from lodash